### PR TITLE
Raise SyntaxError if raw tag is unclosed

### PR DESCRIPTION
**Pull request by pushrax**
See test case.

### CODE REVIEW

File: test/test_helper.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@ def assert_template_result_matches(expected, template, assigns = {}, message = n
       assert_match expected, Template.parse(template).render!(assigns), message
     end
 
-    def assert_match_syntax_error(match, template, registers = {})
+    def assert_match_syntax_error(match, template, assigns = {})
       exception = assert_raises(Liquid::SyntaxError) {
         Template.parse(template).render(assigns)
```
Comment (fw42) [inline test/test_helper.rb]: hu? how did this ever work? or was this method not called anywhere?

**Comment by fw42** [comment]
:+1:

File: test/test_helper.rb (hunk 1 above)
Comment (pushrax) [inline test/test_helper.rb]: It worked because `Template.parse` will raise an exception before `render` is called (and its arguments are evaluated).

**Comment by gauravmc** [comment]
Why not the error when the tag is initialized (like elsewhere) instead of doing it parse time?

**Comment by pushrax** [comment]
@gauravmc this is not quite fixing your original case directly – it's actually ensuring the presence of an `{% endraw %}` following a `{% raw %}`.

**Pull request merged**